### ISSUE TITLE
mAP in VOC2007

### ISSUE THREAD

**Issue by laybebe** [Yejin0111/ADD-GCN#6]
args.seed = 1
args.lr = 0.05
args.image_size = 448
args.batch_size = 16 * 2
args.epoch_step = [30, 40]
the test size is 576
I followed the configuration mentioned above and  used the model that trained on COCO as the pre-train model for Pascal VOC，but best mAP of test of VOC2007 is 94.04%.How can I overcome the problem?

**Comment by Yejin0111** [comment]
> args.seed = 1
> args.lr = 0.05
> args.image_size = 448
> args.batch_size = 16 * 2
> args.epoch_step = [30, 40]
> the test size is 576
> I followed the configuration mentioned above and used the model that trained on COCO as the pre-train model for Pascal VOC，but best mAP of test of VOC2007 is 94.04%.How can I overcome the problem?

I used the model that trained on COCO as the pre-train model for Pascal VOC. You can set ```--resume {Your model path that trained for COCO}``` for voc training.

**Comment by starry097** [comment]
As a new graduate student，I have two problems
1. how can args.batch_size be 16 * 2 ? did you mean you use 2 gpus, or set the batch_size=32?
2. I set "--resume" of the checkpoint on mscoco, do i need to load the "start epoch" from checkpoint?

**Comment by Yejin0111** [comment]
1. Yes.
2. No, only pretrained weights are necessary.

Wang Ziyuan <notifications@github.com> 于2021年1月26日周二 下午8:28写道：

**Comment by starrywintersky** [comment]
Thank you so much !

**Comment by sorrowyn** [comment]
I used ML-GCN model that trained on COCO as the pre-train model for Pascal VOC2007 and the best mAP of test of VOC2007 is 95.58%

**Comment by sorrowyn** [comment]
![74e746fed5cf87e0a296106b9ba397c](https://user-images.githubusercontent.com/19569162/107606147-2fdbc100-6c70-11eb-8b27-30cf803aab88.jpg)

**Comment by sorrowyn** [comment]
![微信截图_20210211135544](https://user-images.githubusercontent.com/19569162/107606380-da53e400-6c70-11eb-9b0a-77d7fe683b24.png)

**Comment by zuiran** [comment]
> I used ML-GCN model that trained on COCO as the pre-train model for Pascal VOC2007 and the best mAP of test of VOC2007 is 95.58%

Hello, sorrowyn. I get the similar result. But i wonder the mAP you got in COCO by MLGCN. With the setting (the way of data augmentation) of this paper, i can just obtain about 83% mAP in MLGCN and a little better in ADDGCN. Do you got the similar results? Thanks.

**Comment by sorrowyn** [comment]
> 
> 
> > I used ML-GCN model that trained on COCO as the pre-train model for Pascal VOC2007 and the best mAP of test of VOC2007 is 95.58%
> 
> Hello, sorrowyn. I get the similar result. But i wonder the mAP you got in COCO by MLGCN. With the setting (the way of data augmentation) of this paper, i can just obtain about 83% mAP in MLGCN and a little better in ADDGCN. Do you got the similar results? Thanks.
On coco, I did not retrain ADDGCN, because I know it is difficult to achieve the results given by the author, the results will always be a little error.

**Comment by zuiran** [comment]
Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directedly?

**Comment by zuiran** [comment]
Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?

**Comment by zjhdxh** [comment]
> Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?

I try ADDGCN on COCO and get 82.6mAP。
![image](https://user-images.githubusercontent.com/55077050/112747054-89f8d180-8fe5-11eb-8249-b39450dc2be8.png)

**Comment by starrywintersky** [comment]
> > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> 
> I try ADDGCN on COCO and get 82.6mAP。
> ![image](https://user-images.githubusercontent.com/55077050/112747054-89f8d180-8fe5-11eb-8249-b39450dc2be8.png)

I got 82.51mAP  without resize to 576

**Comment by zuiran** [comment]
> > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> 
> I try ADDGCN on COCO and get 82.6mAP。
> ![image](https://user-images.githubusercontent.com/55077050/112747054-89f8d180-8fe5-11eb-8249-b39450dc2be8.png)

Thanks for your reply. I am so curious which trick should be made to achieve the mAP presented in this paper.

**Comment by zuiran** [comment]
> > > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> > 
> > 
> > I try ADDGCN on COCO and get 82.6mAP。
> > ![image](https://user-images.githubusercontent.com/55077050/112747054-89f8d180-8fe5-11eb-8249-b39450dc2be8.png)
> 
> I got 82.51mAP without resize to 576

Thanks for your reply

> > > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> > 
> > 
> > I try ADDGCN on COCO and get 82.6mAP。
> > ![image](https://user-images.githubusercontent.com/55077050/112747054-89f8d180-8fe5-11eb-8249-b39450dc2be8.png)
> 
> I got 82.51mAP without resize to 576

Thanks for your share. These experiments on COCO really upset me.

**Comment by sorrowyn** [comment]
![图片](https://user-images.githubusercontent.com/19569162/112774964-f0c1cd80-906d-11eb-99ba-c10cd2c4d41f.png)

**Comment by sorrowyn** [comment]
> 
> 
> Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?

I will try. 
This will take some time.

**Comment by zuiran** [comment]
> > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> 
> I will try.
> This will take some time.

Ok, thanks.

**Comment by sorrowyn** [comment]
> 
> 
> > > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> > 
> > 
> > I will try.
> > This will take some time.
> 
> Ok, thanks.

![图片](https://user-images.githubusercontent.com/19569162/113530043-a1444a00-95f7-11eb-806f-cc543fb00633.png)
![图片](https://user-images.githubusercontent.com/19569162/113530085-bcaf5500-95f7-11eb-82ad-9e5188ea1a85.png)
image size in train is 448 and in  test is 576.
But ML-GCN can only reach 82.x mAP(<82.5)
ADDGCN is an excellent framework.

**Comment by zuiran** [comment]
> > > > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> > > 
> > > 
> > > I will try.
> > > This will take some time.
> > 
> > 
> > Ok, thanks.
> 
> ![图片](https://user-images.githubusercontent.com/19569162/113530043-a1444a00-95f7-11eb-806f-cc543fb00633.png)
> ![图片](https://user-images.githubusercontent.com/19569162/113530085-bcaf5500-95f7-11eb-82ad-9e5188ea1a85.png)
> image size in train is 448 and in test is 576.
> But ML-GCN can only reach 82.x mAP(<82.5)
> ADDGCN is an excellent framework.

Thanks, sorrowyn. By the way, did you train ML-GCN under the framwork of ADDGCN? There are some differences in data augmentation in the part of image cropping which really can boost the performance about 1% mAP used in ADDGCN.

**Comment by zjhdxh** [comment]
> > > > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> > > 
> > > 
> > > I will try.
> > > This will take some time.
> > 
> > 
> > Ok, thanks.
> 
> ![图片](https://user-images.githubusercontent.com/19569162/113530043-a1444a00-95f7-11eb-806f-cc543fb00633.png)
> ![图片](https://user-images.githubusercontent.com/19569162/113530085-bcaf5500-95f7-11eb-82ad-9e5188ea1a85.png)
> image size in train is 448 and in test is 576.
> But ML-GCN can only reach 82.x mAP(<82.5)
> ADDGCN is an excellent framework.

Using this augmentation, ML_GCN can also got 83.9 mAP, ADDGCN is no better than ML_GCN.
By the way, the author obtain the 84.2 in 448 size on README, Can anyone reproduce it?
![image](https://user-images.githubusercontent.com/55077050/113672118-e21c8b80-96e9-11eb-8eaf-2452f2a47136.png)

**Comment by zuiran** [comment]
> > > > > Ok, thanks a lot. Did you try to train ML-GCN or ADDGCN on COCO with the pre-training parameters of backbone (ResNet101) from ImageNet directly?
> > > > 
> > > > 
> > > > I will try.
> > > > This will take some time.
> > > 
> > > 
> > > Ok, thanks.
> > 
> > 
> > ![图片](https://user-images.githubusercontent.com/19569162/113530043-a1444a00-95f7-11eb-806f-cc543fb00633.png)
> > ![图片](https://user-images.githubusercontent.com/19569162/113530085-bcaf5500-95f7-11eb-82ad-9e5188ea1a85.png)
> > image size in train is 448 and in test is 576.
> > But ML-GCN can only reach 82.x mAP(<82.5)
> > ADDGCN is an excellent framework.
> 
> Using this augmentation, ML_GCN can also got 83.9 mAP, ADDGCN is no better than ML_GCN.
> By the way, the author obtain the 84.2 in 448 size on README, Can anyone reproduce it?
> ![image](https://user-images.githubusercontent.com/55077050/113672118-e21c8b80-96e9-11eb-8eaf-2452f2a47136.png)

Yes, thanks. I can not reproduce it too. So I really wonder how to reach it. As well as, I think the backbone ResNet101 with the same setting could also achieve the similar performance comparing to ML-GCN or ADDGCN.

**Comment by Kevinz-code** [comment]
I think the ADDGCN might have use the ResNet-101 pretrained on ImageNet with some data-augmentations (e.g. cut-out or grid mask) that the better pretrained backbone could boost the multi-label's performance, or use a good random seed?
